### PR TITLE
Make service restart (versionfix) safer (launchd)

### PR DESCRIPTION
**Pull request by gabriel**
If fork type is launchd then after calling launchd restart, wait for
the service to startup and be ready before exiting.

Moves the service info path to Env (where it should be).

Also bumps the default timeout for service stop from 2 seconds to 5
seconds.

### CODE REVIEW

**Comment by gabriel** [comment]
cc @maxtaco 

The 1 second sleep which made this work in the past, but I think it's safer to use `libkb.WaitForServiceInfoFile` instead of a sleep?

File: go/client/cmd_launchd_osx.go [hunk 1]
```diff
@@ -326,3 +327,18 @@ func (v *CmdLaunchdAction) Run() error {
 
 	return nil
 }
+
+func RestartLaunchdService(g *libkb.GlobalContext, label string) error {
+	launchService := launchd.NewService(label)
```
Comment (maxtaco) [inline go/client/cmd_launchd_osx.go]: won't this be needed for `keybase ctl restart` too?

File: go/client/versionfix.go [hunk 2]
```diff
@@ -122,18 +121,19 @@ func FixVersionClash(g *libkb.GlobalContext, cl libkb.CommandLine) (err error) {
 	}
 
 	if serviceConfig.ForkType == keybase1.ForkType_LAUNCHD {
-		err = launchd.Restart(serviceConfig.Label, g.Log)
-	} else {
-		ctlCli = keybase1.CtlClient{Cli: gcli}
-		err = ctlCli.Stop(context.TODO(), keybase1.StopArg{})
+		return RestartLaunchdService(g, serviceConfig.Label)
```
Comment (maxtaco) [inline go/client/versionfix.go]: looks good, i like the refactor here to pull the above code out..

**Comment by maxtaco** [comment]
:+1:

File: go/client/cmd_launchd_osx.go (hunk 1 above)
Comment (gabriel) [inline go/client/cmd_launchd_osx.go]: yeah i'll add a PR to call this for `ctl restart` too

**Pull request merged**